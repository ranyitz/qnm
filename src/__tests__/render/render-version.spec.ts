import renderVersion from '../../render/render-version';

it('should colorize the first 10 results', () => {
  const actual = [];

  for (let i = 0; i < 11; i++) {
    actual.push(renderVersion('colorTest', i));
  }

  expect(actual).toMatchSnapshot();

  // no more colors from here
  expect(renderVersion('colorTest', 'bar')).toEqual('bar');
  expect(renderVersion('colorTest', 'bar')).toEqual('bar');
});

it('should start colors for each module from the first color', () => {
  expect(renderVersion('colorEqualityTest', 'foo')).toEqual(
    renderVersion('colorEqualityTest2', 'foo'),
  );
});
