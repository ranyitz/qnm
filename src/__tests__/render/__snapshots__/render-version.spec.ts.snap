// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should colorize the first 10 results 1`] = `
Array [
  "[37m0[39m",
  "[38;5;153m1[39m",
  "[38;5;117m2[39m",
  "[38;5;39m3[39m",
  "[38;5;39m4[39m",
  "[38;5;141m5[39m",
  "[38;5;99m6[39m",
  "[38;5;141m7[39m",
  "[38;5;177m8[39m",
  "[38;5;218m9[39m",
  "[38;5;218m10[39m",
]
`;
